commons/src/test/java/org/eclipse/kapua/commons/util/ResourceUtilsTest.java

```diff
@@ -13,48 +13,112 @@
 
 import java.io.IOException;
 import java.io.Reader;
+import java.lang.reflect.Constructor;
 import java.net.URL;
+import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
 
 import org.eclipse.kapua.qa.markers.junit.JUnitTests;
 import org.junit.Assert;
+import org.junit.ComparisonFailure;
 import org.junit.Test;
 
 import com.google.common.io.CharStreams;
 import org.junit.experimental.categories.Category;
 
 @Category(JUnitTests.class)
-public class ResourceUtilsTest {
+public class ResourceUtilsTest extends Assert {
 
-    /**
-     * Test a successful resource resolve call and a successful read
-     */
     @Test
-    public void testRead1() throws IOException {
-        final URL url = ResourceUtils.getResource("test.properties");
-        Assert.assertNotNull(url);
+    public void testConstructor() throws Exception {
+        Constructor<ResourceUtils> resourceUtils = ResourceUtils.class.getDeclaredConstructor();
+        resourceUtils.setAccessible(true);
+        resourceUtils.newInstance();
+    }
 
-        try (final Reader reader = ResourceUtils.openAsReader(url, StandardCharsets.UTF_8)) {
-            final String string = CharStreams.toString(reader);
-            Assert.assertNotNull(string);
-            Assert.assertFalse(string.isEmpty());
+    @Test
+    public void getResourceTest() {
+        String resource = "test.properties";
+        String noResource = "file:/does-not-exist";
+
+        ClassLoader classloaderTest = Thread.currentThread().getContextClassLoader();
+        URL url = classloaderTest.getResource(resource);
+
+        //Positive test
+        try {
+            assertEquals(url, ResourceUtils.getResource(resource));
+        } catch (AssertionError ex) {
+            fail("AssertionError not expected");
+        }
+
+        //Negative test
+        try {
+            assertEquals(url, ResourceUtils.getResource(noResource));
```

Comment:
If they are not intended to match please use assertNotEquals.
If `ResourceUtils.getResource(...)` is meant to throw an Exception catch the Exception and do asserts on the type of Exception expected.